sql/updates/latest-dev.sql

```diff
@@ -118,3 +118,78 @@ ALTER TABLE IF EXISTS _timescaledb_catalog.compression_chunk_size ADD COLUMN IF
 CLUSTER  _timescaledb_catalog.compression_chunk_size USING compression_chunk_size_pkey;
 ALTER TABLE _timescaledb_catalog.compression_chunk_size SET WITHOUT CLUSTER;
 
+-- Recreate missing dimension slices that might be missing due to a bug
+-- that is fixed in this release. If the dimension slice table is broken and there are dimension
+-- slices missing from the table, we will repair it by:
+--    1. Finding all chunk constraints that have missing dimension
+--       slices and extract the constraint expression from the associated
+--       constraint.
+--    2. Parse the constraint expression and extract the column name,
+--       and upper and lower range values as text.
+--    3. Use the column type to construct the range values (UNIX
+--       microseconds) from these values.
+CREATE OR REPLACE FUNCTION _timescaledb_internal.time_to_internal(time_val ANYELEMENT)
+RETURNS BIGINT AS '@MODULE_PATHNAME@', 'ts_time_to_internal' LANGUAGE C VOLATILE STRICT;
+
+INSERT INTO _timescaledb_catalog.dimension_slice
+WITH
+   -- All dimension slices that are mentioned in the chunk_constraint
+   -- table but are missing from the dimension_slice table.
+   missing_slices AS (
+      SELECT DISTINCT dimension_slice_id
+		    , constraint_name
+		    , attname AS column_name
+		    , pg_get_expr(conbin,conrelid) AS constraint_expr
+	     FROM _timescaledb_catalog.chunk_constraint
+	LEFT JOIN _timescaledb_catalog.dimension_slice sl
+	       ON dimension_slice_id = sl.id
+	     JOIN pg_constraint ON conname = constraint_name
+	     JOIN pg_attribute ON attnum = conkey[1] AND attrelid = conrelid
+	   WHERE dimension_slice_id IS NOT NULL AND sl.id IS NULL
```

Comment:
```suggestion
SELECT
    dimension_slice_id,
    constraint_name,
    attname AS column_name,
    pg_get_expr(conbin, conrelid)
FROM
    _timescaledb_catalog.chunk_constraint cc
    JOIN _timescaledb_catalog.chunk ch ON cc.chunk_id = ch.id
    JOIN pg_constraint ON conname = constraint_name
    JOIN pg_namespace ns ON connamespace = ns.oid
        AND ns.nspname = ch.schema_name
    JOIN pg_attribute ON attnum = conkey[1]
        AND attrelid = conrelid
WHERE
    NOT EXISTS (
        SELECT
            1
        FROM
            _timescaledb_catalog.dimension_slice sl
        WHERE
            dimension_slice_id = sl.id);

```

Reply:
distinct should be avoided, also you were not handling schemas

Reply:
Using `NOT IN` instead. Looks better. Not sure if there is any performance differences, but it does not matter much here.